pkg/component/runtime/manager.go

```diff
@@ -828,20 +833,12 @@ func (m *Manager) waitForStopped(comp *componentRuntimeState) {
 		latestState := comp.getLatest()
 		if latestState.State == client.UnitStateStopped {
 			m.logger.Debugf("component %q stopped.", compID)
-			return
+			return nil
 		}
 
-		m.currentMx.RLock()
-		if _, exists := m.current[compID]; !exists {
-			m.currentMx.RUnlock()
-			return
-		}
-		m.currentMx.RUnlock()
```

Comment:
@michalpristas do you remember why it'd be needed? It seems unnecessary to me. And if no tests fails, I'd believe it indeed is unnecessary